Unable to learn on sliced connections

The following code throws an error in the builder:
```
import nengo
from nengo import spa

import numpy as np
from collections import OrderedDict
import itertools
import ipdb

sim_mode = "gui"
D = 32
rng = np.random.RandomState(0)
vocab = spa.Vocabulary(D, unitary=["ONE"], rng=rng)

vocab.parse("ZERO+ONE+TWO+THREE")

model = spa.SPA(vocabs=[vocab], label="Count Net", seed=0)
n_neurons = 500

with model:
    model.in_1 = spa.State(D)
    model.in_2 = spa.State(D)

    voja1 = nengo.Voja(post_tau=None, learning_rate=5e-2)
    voja2 = nengo.Voja(post_tau=None, learning_rate=5e-2)
    adder = nengo.Ensemble(n_neurons, D*2)
    learning = nengo.Node([0])

    conn_in1 = nengo.Connection(model.in_1.output, adder[D:], learning_rule_type=voja1)
    conn_in2 = nengo.Connection(model.in_2.output, adder[:D], learning_rule_type=voja2)
    nengo.Connection(learning, conn_in1.learning_rule, synapse=None)
    nengo.Connection(learning, conn_in2.learning_rule, synapse=None)

sim = nengo.Simulator(model)
sim.run(1)
```

The error it throws:
```
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
/home/sean/nengo_test/sliced_learning.py in <module>()
     31     nengo.Connection(learning, conn_in2.learning_rule, synapse=None)
     32 
---> 33 sim = nengo.Simulator(model)
     34 sim.run(1)

/home/sean/git/nengo/nengo/simulator.pyc in __init__(self, network, dt, seed, model)
    116 
    117         # Order the steps (they are made in `Simulator.reset`)
--> 118         self.dg = operator_depencency_graph(self.model.operators)
    119         self._step_order = [op for op in toposort(self.dg)
    120                             if hasattr(op, 'make_step')]

/home/sean/git/nengo/nengo/utils/simulator.pyc in operator_depencency_graph(operators)
     45             ups[node].append(op)
     46 
---> 47     validate_ops(sets, ups, incs)
     48 
     49     # -- Scheduling algorithm for serial evaluation:

/home/sean/git/nengo/nengo/utils/simulator.pyc in validate_ops(sets, ups, incs)
     82     # -- assert that only one op sets any particular view
     83     for node in sets:
---> 84         assert len(sets[node]) == 1, (node, sets[node])
     85 
     86     # -- assert that only one op updates any particular view

AssertionError: (Signal(<Ensemble (unlabeled) at 0x7f0aa1531390>.scaled_encoders, shape=(500, 64)), [<nengo.builder.operator.PreserveValue object at 0x7f0aa14ff6d0>, <nengo.builder.operator.PreserveValue object at 0x7f0aa14ffdd0>])

```

Is this a bug or should this be disallowed?

--- a/nengo/builder/connection.py
+++ b/nengo/builder/connection.py
@@ -233,8 +233,11 @@ def build_connection(model, conn):
             targets.append(r.modifies)
 
         if 'encoders' in targets:
-            model.sig[conn.post_obj]['encoders'].readonly = False
-            model.add_op(PreserveValue(model.sig[conn.post_obj]['encoders']))
+            encoder_sig = model.sig[conn.post_obj]['encoders']
+            if not any(isinstance(op, PreserveValue) and op.dst is encoder_sig
+                       for op in model.operators):
+                encoder_sig.readonly = False
+                model.add_op(PreserveValue(encoder_sig))
         if 'decoders' in targets or 'weights' in targets:
             if weights.ndim < 2:
                 raise BuildError(

--- a/nengo/builder/operator.py
+++ b/nengo/builder/operator.py
@@ -69,6 +69,10 @@ class Operator(object):
     def __init__(self, tag=None):
         self.tag = tag
 
+    def __repr__(self):
+        return "<%s%s at 0x%x>" % (
+            self.__class__.__name__, self._tagstr, id(self))
+
     @property
     def _tagstr(self):
         return ' "%s"' % self.tag if self.tag is not None else ''

